### PR TITLE
Completely unpin numpy

### PR DESCRIPTION
**Pull request by OwenArnold**
Fixes https://github.com/scipp/scipp/issues/1700

numpy - boost issue was a red herring. Upgrading numpy on our packages causes no problems. Hard to find problems involving `boost-cpp` `matplotlib-base` was the real problem.

**Tester**

Ignore the seemingly broken CI, Those are just from cancelled jobs picked up from another pipeline.

### CODE REVIEW

File: .azure-pipelines/cron.yml
Comment (OwenArnold) [inline .azure-pipelines/cron.yml:2]: TODO. These changes will need to be reverted after approval.

**Review by matthew-d-jones — APPROVED** [review]
`conda build` runs successfully on my machine on Ubuntu 20.04.

A couple of notes in case I, or someone else, struggles to reproduce this in the future:
It is important to run exactly the same command as in CI:
```
conda build -c conda-forge --python=3.7 .
```
and to do so with conda and conda-build up to date, for example by running `conda build` in a new, clean conda environment.

**Comment by OwenArnold** [comment]
> `conda build` runs successfully on my machine on Ubuntu 20.04.
> 
> A couple of notes in case I, or someone else, struggles to reproduce this in the future:
> It is important to run exactly the same command as in CI
> 
> ```
> conda build -c conda-forge --python=3.7 .
> ```
> 
> and to do so with conda and conda-build up to date, for example by running `conda build` in a new, clean conda environment.

I'm going to add to the scipp deployment docs pages

**Comment by nvaytet** [comment]
Be careful about unpinning matplotlib, as it is pinned in scipp for a good reason: https://github.com/scipp/scipp/pull/1424

**Comment by matthew-d-jones** [comment]
Just so it is documented here:
The problem described in scipp/scipp#1424 which required pinning matplotlib was investigated and found to no longer be an issue. `nodejs 15.11.0`, `matplotlib 3.3.4` and `jupyterlab 3.0.11` were able to be installed and used without issue.

**Pull request merged**
